src/Symfony/Component/Filesystem/Tests/Watcher/Resource/ArrayResourceTest.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Filesystem\Tests\Watcher\Resource;
+
+use Symfony\Component\Filesystem\Tests\FilesystemTestCase;
+use Symfony\Component\Filesystem\Watcher\FileChangeEvent;
+use Symfony\Component\Filesystem\Watcher\Resource\ArrayResource;
+use Symfony\Component\Filesystem\Watcher\Resource\FileResource;
+
+class ArrayResourceTest extends FilesystemTestCase
+{
+    public function testFileChange()
+    {
+        $file = $this->workspace.'/foo.txt';
+        touch($file);
+
+        $resource = new ArrayResource([new FileResource($file)]);
+
+        $this->assertSame([], $resource->detectChanges());
+
+        touch($file, time() + 1);
+
+        $this->assertEquals([new FileChangeEvent($file, FileChangeEvent::FILE_CHANGED)], $resource->detectChanges());
+        $this->assertSame([], $resource->detectChanges());
```

Comment:
> You are testing here behaviour of FileResource, not ArrayResource

I don't understand this part. This is checking that `ArrayResource` correctly returns the detected changes for all the resources defined. The only other option is to create mocks for the changed events here, but I prefer relying on concrete implementations rather than mocking

Reply:
I explained it at 

> Changing return values on subsequent executions of detectChanges is FileResource specific defined behaviour.

This is not something ArrayResource does, so this test case shouldn't explicitly test it. If you want this to be unit test, then indeed you need to provide mock instead of FileResource and assert that ArrayResource calls it when expected, because that's all it does. It won't return different results on subsequent calls for different implementations of decorated resource instance.

Or we can have generic test case like I explain in some other comment, then this is no longer important because there is no ArrayResource specific code, it will just test these things for consistency sake.